src/objectExplorer/objectTypes/treeNodeInfo.ts

```diff
@@ -0,0 +1,266 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import * as vscodeMssql from "vscode-mssql";
+import { NodeInfo } from "../../models/contracts/objectExplorer/nodeInfo";
+import { ObjectExplorerUtils } from "../objectExplorerUtils";
+import * as Constants from "../../constants/constants";
+import { ITreeNodeInfo, ObjectMetadata } from "vscode-mssql";
+import { IConnectionProfile } from "../../models/interfaces";
+
+export class TreeNodeInfo extends vscode.TreeItem implements ITreeNodeInfo {
+    private _nodePath: string;
+    private _nodeStatus: string;
+    private _nodeType: string;
+    private _nodeSubType: string;
+    private _isLeaf: boolean;
+    private _errorMessage: string;
+    private _sessionId: string;
+    private _parentNode: TreeNodeInfo;
+    private _connectionProfile: IConnectionProfile;
+    private _metadata: ObjectMetadata;
+    private _filterableProperties: vscodeMssql.NodeFilterProperty[];
+    private _filters: vscodeMssql.NodeFilter[];
+    private _originalLabel: string;
+
+    public shouldRefresh: boolean = false;
+
+    constructor(
+        label: string,
+        context: vscodeMssql.TreeNodeContextValue,
+        collapsibleState: vscode.TreeItemCollapsibleState,
+        nodePath: string,
+        nodeStatus: string,
+        nodeType: string,
+        sessionId: string,
+        connectionProfile: IConnectionProfile,
+        parentNode: TreeNodeInfo,
+        filterProperties: vscodeMssql.NodeFilterProperty[],
+        nodeSubType: string,
+        objectMetadata?: ObjectMetadata,
+        filters?: vscodeMssql.NodeFilter[],
+    ) {
+        super(label, collapsibleState);
+        this._originalLabel = label;
+        this.context = context;
+        this._nodePath = nodePath;
+        this._nodeStatus = nodeStatus;
+        this._nodeType = nodeType;
+        this._sessionId = sessionId;
+        this._parentNode = parentNode;
+        this._connectionProfile = connectionProfile;
+        this._filterableProperties = filterProperties;
+        this._metadata = objectMetadata;
+        this._filters = filters;
+        this._nodeSubType = nodeSubType;
+        if (this._nodeSubType) {
+            this.iconPath = ObjectExplorerUtils.iconPath(`${this._nodeType}_${this._nodeSubType}`);
+        } else {
+            this.iconPath = ObjectExplorerUtils.iconPath(this.nodeType);
+        }
+        this.id = this.generateId();
+    }
+
+    // Gernating a unique ID for the node
+    protected generateId(): string {
+        return `${this._connectionProfile.id}-${this._nodePath}-${Date.now()}`;
```

Comment:
I saw this ID generation elsewhere; should they be deduped, or does that not matter here?

Reply:
That's just the ConnectionNode overriding generateId since nodepath is not present by default there.